### ISSUE TITLE
Migrations for auth fails on Django 3.1: KeyError: 'deferrable'

### ISSUE THREAD

**Issue by Raekkeri** [ESSolutions/django-mssql-backend#63]
I started a fresh project with Django 3.1. and the latest django-mssql-backend. Running `python manage.py migrate auth` fails on the "0008" migration:

```
  Applying auth.0008_alter_user_username_max_length...Traceback (most recent call last):                           File "manage.py", line 22, in <module>
    main()
  File "manage.py", line 18, in main                                                                                 execute_from_command_line(sys.argv)
...
  File "/.../lib/python3.6/site-packages/django/db/backends/base/schema.py", line 572, in alter_field
    old_db_params, new_db_params, strict)
  File "/.../lib/python3.6/site-packages/sql_server/pyodbc/schema.py", line 479, in _alter_field
    self.execute(self._create_unique_sql(model, columns=[old_field.column]))
  File "/.../lib/python3.6/site-packages/sql_server/pyodbc/schema.py", line 863, in execute
    sql = str(sql)
  File "/.../venv/lib/python3.6/site-packages/django/db/backends/ddl_references.py", line 200, in __str__
    return self.template % self.parts
KeyError: 'deferrable'
```

While experimenting a little, downgrading to Django 2.2 fixed the issue with the problematic migrations, and after upgrading back to 3.1 I was able to run the remaining migrations (those that are new in Django 3). Digging into the code base the problem seems quite straightforward: the `django.db.backends.ddl_references.Statement` seems indeed to expect a `deferrable` keyword argument, and after hacking the source code so that an empty string was given as `deferrable`, the migrations worked just fine. Unfortunately I'm not familiar with mssql or this library, and can't simply attempt a pull request for this.

**Comment by jalpar** [comment]
I have the same issue with Django 3.1.
Migrations work fine with Django 3.0.

I also found this on stackoverflow:

[https://stackoverflow.com/questions/63282351/django-and-azure-sql-key-error-deferrable-when-start-migrate-command](https://stackoverflow.com/questions/63282351/django-and-azure-sql-key-error-deferrable-when-start-migrate-command)

**Comment by CristianH033** [comment]
I had the same problem in Django 3.1 with a custom user model, when I executed the migration I got the following error:

```sh
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length...Traceback (most recent call last):
  File "./manage.py", line 22, in <module>
    main()
  File "./manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/base.py", line 85, in wrapped
    res = handle_func(*args, **kwargs)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 243, in handle
    post_migrate_state = executor.migrate(
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/migrations/operations/fields.py", line 236, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 571, in alter_field
    self._alter_field(model, old_field, new_field, old_type, new_type,
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/sql_server/pyodbc/schema.py", line 479, in _alter_field
    self.execute(self._create_unique_sql(model, columns=[old_field.column]))
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/sql_server/pyodbc/schema.py", line 861, in execute
    sql = str(sql)
  File "/home/cristian/.virtualenvs/SIM2/lib/python3.8/site-packages/django/db/backends/ddl_references.py", line 200, in __str__
    return self.template % self.parts
KeyError: 'deferrable'
```
Downgrading to Django 3.0.x made the migration work properly.

**Comment by shakthifuture** [comment]
Any solution? I need to use Django 3.1, I can't downgrade

**Comment by franzejr** [comment]
Got the same error on Django `3.1.1`.

**Comment by franzejr** [comment]
I confirm downgrading to Django 3.0.10 fixes the issue.

**Comment by tietjen-douglas** [comment]
I have also come across this issue with Django 3.1.  I hope a fix can come soon.

**Comment by mriffey** [comment]
I also see this issue.

**Comment by ProjectBreeze** [comment]
Bumping this issue up, I would love to use Django 3.1 with this.  Thanks!

**Comment by EthanCundick** [comment]
Same problem for me on Django 3.1.4

**Comment by kumbhagirishiva** [comment]
Same problem for me on Django 3.1.4
Is their any solution to this, I was downgraded to 3.0.6 also but not solved.
Any body knows about this error ,please give solution

**Comment by kumbhagirishiva** [comment]
> Same problem for me on Django 3.1.4

If you solved this error, please give a solution to me

**Comment by kumbhagirishiva** [comment]
> Any solution? I need to use Django 3.1, I can't downgrade

If you solved this error, please give a solution to me

**Comment by Lazloo** [comment]
Same problem on my site but, only when I use django-mssql-backend

**Comment by dcleveng** [comment]
I'm using django-mssql-backend and ran in to this issue on two separate occasions. Downgrading fixed the issue the first time around, but here's what I did to work around it the second time: 

I found that removing **unique=True** from the migration file on one of the fields I was modifying allowed the migration to proceed without downgrading Django.

**Comment by OskarPersson** [comment]
Closed in #86

**Comment by BearGaming** [comment]
Issue still persists in Django 3.2
